airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -1600,6 +1602,74 @@ public boolean getConnectionHasAlphaOrBetaConnector(final UUID connectionId) thr
     return countResult > 0;
   }
 
+  public Optional<ConnectorBuilderProject> getConnectorBuilderProject(final UUID builderProjectId, final UUID workspaceId) throws IOException {
+    return database.query(ctx -> ctx.select(CONNECTOR_BUILDER_PROJECT.asterisk())
+        .from(CONNECTOR_BUILDER_PROJECT)
+        .where(getBuilderProjectIdCondition(builderProjectId, workspaceId))
+        .fetch())
+        .map(DbConverter::buildConnectorBuilderProject)
+        .stream()
+        .findFirst();
+  }
+
+  public Stream<ConnectorBuilderProject> getConnectorBuilderProjectsByWorkspace(final UUID workspaceId) throws IOException {
+    final Condition matchByIds = CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID.eq(workspaceId);
+
+    return database
+        .query(ctx -> ctx
+            .select(CONNECTOR_BUILDER_PROJECT.ID, CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID, CONNECTOR_BUILDER_PROJECT.NAME,
```

Comment:
I don't see where `getConnectorBuilderProjectsByWorkspace` is used but is this clear to the user that `ConnectorBuilderProject` can be partial?

Reply:
It's used in a follow-up PR - it should be clear from the context, having a separate type for that (`ConnectorBuilderProject` vs `ConnectorBuilderProjectWithManifest`) seemed over-engineered to me, but no strong opinion.